### PR TITLE
Generate Allure results in real-time

### PR DESCRIPTION
**Pull request by dtinth**
### Context
The original implementation

- Generates Allure tests results data only when test is finished.
- Has some invalid methods (e.g. `onTimeout`).

This PR upgrades Playwright Test to latest version to reflect latest Playwright API. All tests are passing.

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: packages/allure-playwright/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (just-boris) [inline packages/allure-playwright/src/index.ts:1]: why is this needed? There are `eslint-disable-line` below

File: packages/allure-playwright/src/index.ts [hunk 2]
```diff
@@ -17,130 +18,160 @@
...
+        console.log(JSON.stringify({ error: (e as Error).stack || String(e) })); // eslint-disable-line no-console
       }
     }
   }
+
+  private getAllureGroup(suite: Suite): AllureGroup {
+    let group = this.allureGroupCache.get(suite);
+    if (!group) {
+      const parent = suite.parent ? this.getAllureGroup(suite.parent) : this.allureRuntime;
+      group = parent.startGroup(suite.title);
```
Comment (just-boris) [inline packages/allure-playwright/src/index.ts:160]: It is not expected that `get*` methods have any side effects. Better to call it `ensureGroupStarted`

File: packages/allure-playwright/src/index.ts [hunk 3]
```diff
@@ -17,130 +18,160 @@
...
+      this.allureGroupCache.set(suite, group);
+    }
+    return group;
+  }
+
+  private getAllureStep(step: TestStep, allureTest: AllureTest): AllureStep {
+    let allureStep = this.allureStepCache.get(step);
+    if (!allureStep) {
+      const parent = step.parent ? this.getAllureStep(step.parent, allureTest) : allureTest;
+      allureStep = parent.startStep(step.title);
```
Comment (just-boris) [inline packages/allure-playwright/src/index.ts:170]: Same comment about `get*` name

**Review by just-boris — APPROVED** [review]
The implementation looks good, but there are a few stylistic suggestions

File: packages/allure-playwright/src/index.ts (hunk 1 above)
Comment (dtinth) [reply packages/allure-playwright/src/index.ts:1]: Oops, my bad 🤪

**Comment by dtinth** [comment]
@just-boris Thank you for your comment! My original intention was to allow Playwright tests to invoke Allure APIs directly to add attachments and labels. I got stuck where when I realized that Playwright runs tests and reporters in different Node process.

I found that another PR is already solving my problem:

- https://github.com/allure-framework/allure-js/pull/408

IMO, I would prefer having that feature in first. Maybe I can rebase my changes on top of that after it got merged (it seems like our changes are not compatible with each other yet).

**Comment by vovsemenv** [comment]
i love it 💚

**Comment by baev** [comment]
> I found that another PR is already solving my problem

@dtinth  we decided to do the opposite, as #408 is still waiting for playwright 1.18 release and your one is good to go

**Comment by baev** [comment]
@dtinth could you please rebase your PR?

**Comment by skhomuti** [comment]
@dtinth are you still interested in this pull request?

**Comment by dtinth** [comment]
Yes — sorry, I got caught up in other things 😞, will update it. Thanks for the kind reminder!

**Comment by dtinth** [comment]
Rebased & review comments have been addressed.

**Comment by baev** [comment]
thanks! 🎉
